hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/MockNodeManager.java

```diff
@@ -309,6 +309,10 @@ public List<DatanodeDetails> getNodes(
       return deadNodes;
     }
 
+    if (nodestate == null) {
```

Comment:
What is the case nodestate is null?

Reply:
oh it's because your passing null when calling getNodeCount?
That means null makes it as any state.

Reply:
instead of passing null, getAllNodes().size() also works?